cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -2429,6 +2430,13 @@ void RegisterScalarArithmetic(FunctionRegistry* registry) {
     DCHECK_OK(subtract->AddKernel({in_type, in_type}, duration(unit), std::move(exec)));
   }
 
+  // Add subtract(duration, duration) -> duration
```

Comment:
I don't see any tests for this, do we need to add some?

Reply:
Because subtraction can be done directly on times I removed this.